ddtrace/profiling/exporter/pprof.pyx

```diff
@@ -281,53 +322,87 @@ class _PprofConverter(object):
         )
 
 
+_stack_event_group_key_T = typing.Tuple[
+    int,
+    int,
+    str,
+    typing.Optional[int],
+    typing.Optional[int],
+    typing.Tuple,
+    int,
+]
+
+
 @attr.s
 class PprofExporter(exporter.Exporter):
     """Export recorder events to pprof format."""
 
     @staticmethod
-    def _get_trace_id(event):
-        if event.trace_id:
-            return str(event.trace_id)
-        return ""
-
-    @staticmethod
-    def _get_span_id(event):
-        if event.span_id:
-            return str(event.span_id)
-        return ""
+    def _none_to_str(
+        value,  # type: typing.Optional[typing.Any]
+    ):
+        # type: (...) -> str
+        if value is None:
+            return ""
+        return str(value)
 
     @staticmethod
     def _get_thread_name(thread_id, thread_name):
         if thread_name is None:
             return "Anonymous Thread %d" % thread_id
         return thread_name
 
-    def _stack_event_group_key(self, event):
-        # If multiple traces were active, we pick only one :(
+    def _stack_event_group_key(
+        self,
+        event,  # type: stack.StackSampleEvent
+    ):
+        # type: (...) -> _stack_event_group_key_T
+
+        if event.trace_type == ext.SpanTypes.WEB.value:
+            trace_resource = self._none_to_str(event.trace_resource)
+        else:
+            # Do not export trace_resource for privacy concerns.
+            trace_resource = ""
+
         return (
             event.thread_id,
             event.thread_native_id,
             self._get_thread_name(event.thread_id, event.thread_name),
-            self._get_trace_id(event),
-            self._get_span_id(event),
+            self._none_to_str(event.trace_id),
+            self._none_to_str(event.span_id),
+            trace_resource,
+            self._none_to_str(event.trace_type),
             tuple(event.frames),
             event.nframes,
         )
 
-    def _group_stack_events(self, events):
+    def _group_stack_events(
+        self,
+        events,  # type: typing.Iterable[stack.StackSampleEvent]
+    ):
+        # type: typing.Iterator[typing.Tuple[_stack_event_group_key_T, typing.Iterator[stack.StackSampleEvent]]]
         return itertools.groupby(
             sorted(events, key=self._stack_event_group_key),
             key=self._stack_event_group_key,
         )
 
-    def _lock_event_group_key(self, event):
+    def _lock_event_group_key(
+        self, event  # type: lock.LockEventBase
+    ):
+        if event.trace_type == ext.SpanTypes.WEB.value:
+            trace_resource = self._none_to_str(event.trace_resource)
```

Comment:
there is likely an edge case with a few integrations where we don't set the resource until after the request has finished.

Pylons does this, not sure how many others, but could imagine if we don't get the resource _before_ creating the span, could still be a race condition on whether we have it or not

Reply:
Right, though I am not sure this should block this PR?
That sounds like it's something we want to "improve" as possible in integrations.